kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -191,6 +212,71 @@ private void handleMetadata(ColumnVector metadataVector) {
     }
   }
 
+  /**
+   * Checks whether each of the current transaction's {@link DomainMetadata} conflicts with the
+   * winning transaction at any domain.
+   *
+   * <ol>
+   *   1. Accept the current transaction if its set of metadata domains does not overlap with the
+   *   winning transaction's set of metadata domains.
+   *   <p>2. Otherwise, fail the current transaction unless each conflicting domain is associated
+   *   with a domain-specific way of resolving the conflict.
+   * </ol>
+   *
+   * @param actionBatch action batch from the winning transactions
+   */
+  private void handleDomainMetadata(ColumnarBatch actionBatch) {
+    // Extract the domain metadata map from the winning transaction.
+    Map<String, DomainMetadata> winningDomainMetadataMap =
+        DomainMetadataUtils.extractDomainMetadataMap(
+            inMemoryIterable(actionBatch.getRows()), CONFLICT_RESOLUTION_SCHEMA);
+
+    // Get the ordinal of the domainMetadata action from the full schema, which is used when writing
+    // out the single action to the Delta Log
+    final int domainMetadataOrdinal = FULL_SCHEMA.indexOf("domainMetadata");
+
+    // Use try-with-resources to ensure that the CloseableIterable is closed after the loop
+    try (CloseableIterable<Row> closeableDataActions = dataActions) {
+      for (Row action : closeableDataActions) {
+        // Skip non-domainMetadata actions
+        if (action.isNullAt(domainMetadataOrdinal)) continue;
+
+        // Extract DomainMetadata action that the losing transaction trying to commit
+        DomainMetadata domainMetadata =
+            DomainMetadata.fromRow(action.getStruct(domainMetadataOrdinal));
+
+        // Try to resolve the conflict, if not possible, throw a ConcurrentTransaction exception
+        resolveDomainMetadataConflict(domainMetadata, winningDomainMetadataMap);
+      }
+    } catch (IOException ioe) {
+      throw new UncheckedIOException(ioe);
+    }
+  }
+
+  private void resolveDomainMetadataConflict(
+      DomainMetadata domainMetadataAttempt, Map<String, DomainMetadata> winningDomainMetadataMap) {
+
+    String domain = domainMetadataAttempt.getDomain();
+    DomainMetadata winningDomainMetadata = winningDomainMetadataMap.get(domain);
+    if (winningDomainMetadata == null) {
+      // No conflict
+      return;
```

Comment:
you can just invert this `if` statement, if winningDomainMetadata != null then throw

Reply:
Updated.